boa/src/value/mod.rs

```diff
@@ -32,6 +32,7 @@ mod operations;
 mod rcbigint;
 mod rcstring;
 mod rcsymbol;
+mod to_integer_or_infinity;
```

Comment:
I think there is no need to put this in a separate file, like we do with the other `Value::to_*` functions.